pkg/generator/appconfiguration/generator/workload/network/ports_generator.go

```diff
@@ -113,6 +116,9 @@ func (g *portsGenerator) validate() error {
 	if len(g.selector) == 0 {
 		return ErrEmptySelectors
 	}
+	if len(g.ports) == 0 {
```

Comment:
zero ports is allowed according to our recent discussion

Reply:
If empty ports, there will be portsGenerator, which is done in service_generator.go
Delete validation inside the generator is not cool and make others confused, I fix it back from https://github.com/KusionStack/kusion/pull/510. I have already replied to it, and more serious review is needed. @liu-hm19

Reply:
What do you mean that portsGenerator is done in service_generator.go. Can you provide the specified code?